### PR TITLE
TYP: use type annotations in tzconversion.pyi

### PR DESCRIPTION
**Pull request by fangchenli**

### CODE REVIEW

File: pandas/_libs/tslibs/tzconversion.pyi [hunk 1]
```diff
@@ -1,25 +1,21 @@
+from __future__ import annotations
```
Comment (simonjayhawkins) [inline pandas/_libs/tslibs/tzconversion.pyi:1]: maybe not needed for stub file. will need to check.

Comment (jreback) [reply pandas/_libs/tslibs/tzconversion.pyi:1]: don't we need for the `|` syntax?

Comment (simonjayhawkins) [reply pandas/_libs/tslibs/tzconversion.pyi:1]: a .pyi is not parsed when running pandas. It is only used by mypy (or other type checkers). The syntax of a .pyi file is not constrained by the minimum python interpreter supported by pandas.

Comment (simonjayhawkins) [reply pandas/_libs/tslibs/tzconversion.pyi:1]: > will need to check.

running pre-commit we don't have any rules that prohibit this.

Comment (simonjayhawkins) [reply pandas/_libs/tslibs/tzconversion.pyi:1]: > > will need to check.
> 
> running pre-commit we don't have any rules that prohibit doing this.

**Pull request merged**